ngraph/frontend/tensorflow/src/decoder_proto.cpp

```diff
@@ -0,0 +1,121 @@
+        longs.reserve(attrs[0].list().i_size());
+        for (size_t idx = 0; idx < attrs[0].list().i_size(); ++idx) {
+            longs.push_back(attrs[0].list().i(idx));
+        }
+        return std::make_shared<VariantWrapper<std::vector<int64_t>>>(longs);
+    } else if (type_info == VariantWrapper<int32_t>::type_info) {
+        return std::make_shared<VariantWrapper<int32_t>>(static_cast<int32_t>(attrs[0].i()));
+    } else if (type_info == VariantWrapper<std::vector<int32_t>>::type_info) {
+        std::vector<int32_t> ints;
+        ints.reserve(attrs[0].list().i_size());
+        for (size_t idx = 0; idx < attrs[0].list().i_size(); ++idx) {
+            ints.push_back(static_cast<int32_t>(attrs[0].list().i(idx)));
+        }
+        return std::make_shared<VariantWrapper<std::vector<int32_t>>>(ints);
+    } else if (type_info == VariantWrapper<float>::type_info) {
+        return std::make_shared<VariantWrapper<float>>(attrs[0].f());
+    } else if (type_info == VariantWrapper<std::vector<float>>::type_info) {
+        std::vector<float> floats;
+        floats.reserve(attrs[0].list().i_size());
+        for (size_t idx = 0; idx < attrs[0].list().i_size(); ++idx) {
+            floats.push_back(attrs[0].list().f(idx));
+        }
+        return std::make_shared<VariantWrapper<std::vector<float>>>(floats);
+    } else if (type_info == VariantWrapper<ngraph::element::Type>::type_info) {
+        auto data_type = attrs[0].type();
+        return std::make_shared<VariantWrapper<ngraph::element::Type>>(TYPE_MAP[data_type]);
+    } else if (type_info == VariantWrapper<bool>::type_info) {
+        return std::make_shared<VariantWrapper<bool>>(attrs[0].b());
+    } else if (type_info == VariantWrapper<::tensorflow::DataType>::type_info) {
+        return std::make_shared<VariantWrapper<::tensorflow::DataType>>(attrs[0].type());
+    } else if (type_info == VariantWrapper<::tensorflow::TensorProto>::type_info) {
+        return std::make_shared<VariantWrapper<::tensorflow::TensorProto>>(attrs[0].tensor());
+    } else if (type_info == VariantWrapper<::ngraph::PartialShape>::type_info) {
+        std::vector<ngraph::Dimension> dims;
+        auto tf_shape = attrs[0].shape();
+        for (int i = 0; i < tf_shape.dim_size(); i++) {
+            dims.push_back(tf_shape.dim(i).size());
+        }
+        auto pshape = ngraph::PartialShape(dims);
+        return std::make_shared<VariantWrapper<::ngraph::PartialShape>>(pshape);
+    }
+
+    // type is not supported by decoder
+    return nullptr;
+}
+
+size_t DecoderTFProto::get_input_size() const {
+    return m_node_def->input_size();
+}
+
+void DecoderTFProto::get_input_node(const size_t input_port_idx,
+                                    std::string& producer_name,
+                                    size_t& producer_output_port_index) const {
+    std::string producer_port_name = m_node_def->input(input_port_idx);
+    auto delim_pos = producer_port_name.find(':');
```

Comment:
We never have a situation when there are two ":" characters in the `producer_name`?